.github/workflows/native.yml

```diff
@@ -0,0 +1,223 @@
+  ccache_dir: "${{ github.workspace }}/.ccache"
+  ccache_compilercheck: content
+  cache_compress: 'true'
+  ccache_maxsize: 50M
+  ccache_cmake: -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_C_COMPILER_LAUNCHER=ccache
+jobs:
+  ubuntu:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - name: Ubuntu 18.04 full
+          os: ubuntu-18.04
+          identifier: ubuntu_1804_full
+          cmake: -DCOMPILE_TESTS=on
+          brt_tags: ''
+          unittests: 'true'
+        - name: Ubuntu 18.04 minimal
+          os: ubuntu-18.04
+          identifier: ubuntu_1804_minimal
+          cmake: -DCOMPILE_TESTS=off -DUSE_WASM_COMPATIBLE_SOURCE=on
+          brt_tags: "'#wasm'"
+          unittests: 'false'
+        - name: Ubuntu 20.04 full
+          os: ubuntu-20.04
+          identifier: ubuntu_2004_full
+          cmake: -DCOMPILE_TESTS=on
+          brt_tags: ''
+          unittests: 'true'
+        - name: Ubuntu 20.04 minimal
+          os: ubuntu-20.04
+          identifier: ubuntu_2004_minimal
+          cmake: -DCOMPILE_TESTS=off -DUSE_WASM_COMPATIBLE_SOURCE=on
+          brt_tags: "'#wasm'"
+          unittests: 'false'
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - name: Install Dependencies
+      run: |-
+        sudo apt-get update
+        sudo apt-get install -y \
+          libgoogle-perftools-dev libprotobuf-dev protobuf-compiler \
+          libboost-all-dev ccache
+    - name: Install MKL
+      run: |-
+        wget -qO- "https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2019.PUB" | sudo apt-key add -
+        sudo sh -c "echo deb https://apt.repos.intel.com/mkl all main > /etc/apt/sources.list.d/intel-mkl.list"
+        sudo apt-get update -o Dir::Etc::sourcelist="/etc/apt/sources.list.d/intel-mkl.list"
+        sudo apt-get install -y --no-install-recommends intel-mkl-64bit-2020.0-088
+    - name: Generate ccache_vars for ccache based on machine
+      shell: bash
+      id: ccache_vars
+      run: |-
+        echo "::set-output name=hash::$(echo ${{ env.ccache_compilercheck }})"
+        echo "::set-output name=timestamp::$(date '+%Y-%m-%dT%H.%M.%S')"
+    - name: Cache-op for build-cache through ccache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.ccache_dir }}
+        key: ccache-${{ matrix.identifier }}-${{ steps.ccache_vars.outputs.hash }}-${{ github.ref }}-${{ steps.ccache_vars.outputs.timestamp }}
+        restore-keys: |-
+          ccache-${{ matrix.identifier }}-${{ steps.ccache_vars.outputs.hash }}-${{ github.ref }}
+          ccache-${{ matrix.identifier }}-${{ steps.ccache_vars.outputs.hash }}
+          ccache-${{ matrix.identifier }}
+    - name: ccache environment setup
+      run: |-
+        echo "CCACHE_COMPILER_CHECK=${{ env.ccache_compilercheck }}" >> $GITHUB_ENV
```

Comment:
Is there any reason we can't use the proper names for these when defining `env:` and skip this step?

Reply:
Are both `env.{xx}` and environment variable  (bash / PS / etc) same? I'm treating `env` (global, job), `matrix` more as variable store than environment variables updated behind the scenes in YAML generation, and this notation is more explicit.

> You can use most contexts at any point in your workflow, including when default environment variables would be unavailable.  

https://docs.github.com/en/actions/reference/environment-variables#determining-when-to-use-default-environment-variables-or-contexts

Reply:
My experience with Ubuntu/MacOS is that the name has been the same. The `env` context contains only the items defined in the YAML entry, while for `run: env` it contains items in the context as well as the default runner environment.

If I want to add a new CCACHE option, it's clearer and easier to do it once.

Reply:
Discussed and agreed upon offline that we're keeping the additional indirection, to maintain the ops to be more explicit and consistent.